scripts/bootstrap.sh

```diff
@@ -14,28 +14,77 @@
 # limitations under the License.
 #
 
-if [[ -n ${BASH_SOURCE[0]} ]]; then
-    CHIP_ROOT=$(cd "${BASH_SOURCE[0]%/*}/.." && pwd)
-else
-    CHIP_ROOT=$(cd "${0%/*}/.." && pwd)
-fi
+_bootstrap_or_activate() {
+    if [ -n "$BASH" ]; then
+        local _BOOTSTRAP_PATH="${BASH_SOURCE[0]}"
+    else
+        local _BOOTSTRAP_PATH="$1"
+    fi
 
-export PW_BRANDING_BANNER="$CHIP_ROOT/.chip-banner.txt"
-export PW_BRANDING_BANNER_COLOR="bold_white"
-export PW_VIRTUALENV_REQUIREMENTS="$CHIP_ROOT/scripts/requirements.txt"
+    local _BOOTSTRAP_NAME="${_BOOTSTRAP_PATH##*/}"
+    local _CHIP_ROOT="$(cd "${_BOOTSTRAP_PATH%/*}/.." && pwd)"
 
-if test -f "$CHIP_ROOT/scripts/helpers/rename_submodules.sh"; then
-    "$CHIP_ROOT/scripts/helpers/rename_submodules.sh"
-fi
+    if [ "$_BOOTSTRAP_NAME" = "bootstrap.sh" ]; then
+        git submodule update --init
+    fi
 
-git submodule update --init
+    local _CHIP_BANNER="$(
+        cat <<EOF
+  ▄███▒  ░▓█  ░▓█ ░▓█▓ ▒█████▄
+ ██▒ ▀█▒  ▒█   ▒█  ░█▒  ▒█░  █░
+ █▓░      ▒██████  ░█▒  ▒█▄▄▄█░
+ ▓█   █▒  ▒█   ▒█  ░█░  ▒█▀
+ ░▓███▀  ░▓███░▓█▒ ░█░  ▒█
+EOF
+    )"
 
-export PW_VIRTUALENV_SETUP_PY_ROOTS="$CHIP_ROOT/integrations/mobly"
+    PW_ROOT="$_CHIP_ROOT/third_party/pigweed/repo"
+    export PW_ROOT
 
-# shellcheck source=/dev/null
-source "$CHIP_ROOT/third_party/pigweed/repo/bootstrap.sh"
+    . "$_CHIP_ROOT/third_party/pigweed/repo/pw_env_setup/util.sh"
 
-#TODO - remove this once native python building is solved for
-#       psutil (one of mobly's dependencies which CHIP does
-#       not actually need, so --no-deps is OK)
-pip install --no-deps portpicker mobly
+    _chip_bootstrap_banner() {
+        if [ -z "$PW_ENVSETUP_QUIET" ] && [ -z "$PW_ENVSETUP_NO_BANNER" ]; then
```

Comment:
why does pigweed ask for _**2**_ environment variables to be set to get it to be quiet?

Reply:
It's the other way around, if neither are set it will print.